operator/deploy/deploy-kiali-operator.sh

```diff
@@ -398,36 +394,37 @@ echo "Deploying Kiali CR to namespace [${OPERATOR_NAMESPACE}]"
 build_spec_value() {
   local var_name=${1}
   local var_value=${!2-_undefined_}
-  if [ "${var_value}" == "_undefined_" ]; then
-    echo
+  local var_show_empty=${3:-false}
+  if [ "${var_value}" == "_undefined_" -a "${var_show_empty}" == "false" ]; then
+    return
   else
-    if [ "${var_value}" == "" ]; then
+    if [ "${var_value}" == "" -o "${var_value}" == "_undefined_" ]; then
       var_value='""'
     fi
     echo "$var_name: $var_value"
   fi
 }
 
 cat <<EOF | ${CLIENT_EXE} apply -n ${OPERATOR_NAMESPACE} -f -
-apiVersion: op.kiali.io/v1alpha1
+apiVersion: kiali.io/v1alpha1
 kind: Kiali
 metadata:
   name: kiali
 spec:
-  $(build_spec_value auth_strategy AUTH_STRATEGY)
-  $(build_spec_value grafana_url GRAFANA_URL)
-  $(build_spec_value image_name IMAGE_NAME)
-  $(build_spec_value image_pull_policy IMAGE_PULL_POLICY)
-  $(build_spec_value image_version IMAGE_VERSION)
   $(build_spec_value istio_namespace ISTIO_NAMESPACE)
```

Comment:
@jmazzitelli ISTIO_NAMESPACE used here does not defaults to istio-system as described in _Environment variables that affect Kiali_ section

Reply:
@FilipB while it isn't clear, Kiali will actually default to "istio-system".

Here you will notice the script will NOT put the "istio_namespace" setting in the CR because you did not provide it to the script. This tells Kiali to set the istio namespace to the default of "istio-system".

When the comments at the top of the script document the defaults, it is documenting the final defaults that Kiali uses if you do not specify them to the script. In this case, the deploy script is not setting it in the CR because you (the user) did not provide it - so ultimately Kiali will default to "istio-system".

Reply:
Ah, right. build_spec_value function checks if ISTIO_NAMESPACE is empty. Thanks.